### PR TITLE
Improve release automation with Asana magic

### PR DESCRIPTION
**Pull request by GioSensation**
**Reviewers:** @jonathanKingston @MariagraziaAlastra @sammacbeth 
**Asana:** https://app.asana.com/0/0/1200750505563921/1201183123265449/f

## Description:
Automate the Asana paperwork!

- Run the workflow, with manual input of version (as the current stage.yml)
- Run the script and commit the changes
- Generate a new Release draft (prerelease: true)
- Use the list of commits from the previous tag to the current to populate the release notes
- Build zip files and upload them to the release ← We already have most of this logic in publish.yml
- Asana Script:
  - Get list of tasks in the Extension release project under the Release section
  - Duplicate the release template, paste in the list of tasks in the appropriate section
  - Upload the zip files to Asana

## Steps to test this PR:
Give it a first pass, then we'll merge and try to cut test releases to see how it works. You can see what the Asana script spits out in https://app.asana.com/0/312629933896096/1201202709634481/f.

### CODE REVIEW

File: .github/workflows/stage.yml [hunk 1]
```diff
@@ -10,25 +10,75 @@ on:
...
+      - uses: actions/cache@v2
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+        - name: Install dependencies
+          run: |
+            npm ci
+            npm i -g web-ext
```
Comment (sammacbeth) [inline .github/workflows/stage.yml:53]: It looks like we're running `Install dependencies` twice in the same pipeline. Is that on purpose?

File: .github/workflows/stage.yml [hunk 2]
```diff
@@ -10,25 +10,75 @@ on:
...
+              ./build/firefox/release/web-ext-artifacts/*.zip
+              ./build/chrome/release/*.zip
+
+        - name: Asana Workflow
+            env:
+              ASANA_ACCESS_TOKEN: ${{ secrets.ASANA_ACCESS_TOKEN }}
+              VERSION: ${{ github.event.release.tag_name }}
+              RELEASE_URL: ${{ github.event.release.html_url }}
+            run: |
+              npm install
```
Comment (sammacbeth) [inline .github/workflows/stage.yml:83]: Way do we need to run `npm install` again?

**Review by sammacbeth — COMMENTED** [review]
Looks good. It seems we're running npm install too often, but otherwise this is great! Also needs a rebase, but then feel free to merge.

File: .github/workflows/stage.yml (hunk 2 above)
Comment (GioSensation) [reply .github/workflows/stage.yml:83]: Just oversight. I'm removing the extra installs. While checking, I noticed that we also have an `install-ci` command in package.json (see [here](https://github.com/duckduckgo/duckduckgo-privacy-extension/blob/2e0473ba17a9314c63fdfd48aac728b26da9d57a/package.json#L10)). Should we use that instead? It rebuilds puppeteer (which we don't need) but I'm not sure about the rest. cc @jonathanKingston

Comment (sammacbeth) [reply .github/workflows/stage.yml:83]: Yes I think we should use `install-ci` as this prevents a potential exploit of the build via npm install scripts.

File: .github/workflows/stage.yml [hunk 3]
```diff
@@ -10,25 +10,70 @@ on:
...
+        uses: metcalfc/changelog-generator@c3f61b2a3db543ceee83c759ac8e48419a1a2211
+        with:
+          myToken: ${{ secrets.GITHUB_TOKEN }}
+
+      - uses: actions/cache@v2
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
```
Comment (sammacbeth) [inline .github/workflows/stage.yml:49]: This should be moved to before the `install-ci` step (as in https://github.com/duckduckgo/duckduckgo-privacy-extension/blob/develop/.github/workflows/tests.yml#L14-L20). This is step restores a cached `.npm` folder for so the `npm install` artifacts don't need to be fetched.

See https://github.com/actions/cache/blob/main/examples.md#node---npm for more on how this cache step works.

**Review by sammacbeth — CHANGES_REQUESTED** [review]

File: .github/workflows/stage.yml (hunk 3 above)
Comment (GioSensation) [reply .github/workflows/stage.yml]: Thanks for noticing. I've pushed the change. Do you mind checking again if I got it right this time 🙂?

**Review by sammacbeth — APPROVED** [review]

**Pull request merged**